jetty-server/src/test/java/org/eclipse/jetty/server/ForwardedRequestCustomizerTest.java

```diff
@@ -497,7 +497,67 @@ public void destroy() throws Exception
                     .requestURL("http://fw.example.com:4333/")
                     .remoteAddr("8.5.4.3").remotePort(2222)
             ),
-
+            Arguments.of(new Request("X-Forwarded-* (Multiple Ports)")
+                    .headers(
+                        "GET / HTTP/1.1",
+                        "Host: myhost:10001",
+                        "X-Forwarded-For: 127.0.0.1:8888,127.0.0.2:9999",
+                        "X-Forwarded-Port: 10002",
+                        "X-Forwarded-Proto: https",
+                        "X-Forwarded-Host: sub1.example.com:10003",
+                        "X-Forwarded-Server: sub2.example.com"
+                    ),
+                new Expectations()
+                    .scheme("https").serverName("sub1.example.com").serverPort(10002)
+                    .requestURL("https://sub1.example.com:10002/")
+                    .remoteAddr("127.0.0.1").remotePort(8888)
+            ),
+            Arguments.of(new Request("X-Forwarded-* (Multiple Ports - Server First)")
+                    .headers(
+                        "GET / HTTP/1.1",
+                        "X-Forwarded-Server: sub2.example.com:10007",
+                        "Host: myhost:10001",
+                        "X-Forwarded-For: 127.0.0.1:8888,127.0.0.2:9999",
+                        "X-Forwarded-Proto: https",
+                        "X-Forwarded-Port: 10002",
+                        "X-Forwarded-Host: sub1.example.com:10003"
+                    ),
+                new Expectations()
+                    .scheme("https").serverName("sub1.example.com").serverPort(10002)
+                    .requestURL("https://sub1.example.com:10002/")
+                    .remoteAddr("127.0.0.1").remotePort(8888)
+            ),
+            Arguments.of(new Request("X-Forwarded-* (Multiple Ports - setForwardedPortAsAuthority = false)")
+                    .configureCustomizer((customizer) -> customizer.setForwardedPortAsAuthority(false))
+                    .headers(
+                        "GET / HTTP/1.1",
+                        "Host: myhost:10001",
+                        "X-Forwarded-For: 127.0.0.1:8888,127.0.0.2:9999",
+                        "X-Forwarded-Port: 10002",
+                        "X-Forwarded-Proto: https",
+                        "X-Forwarded-Host: sub1.example.com:10003",
+                        "X-Forwarded-Server: sub2.example.com"
+                    ),
+                new Expectations()
+                    .scheme("https").serverName("sub1.example.com").serverPort(10003)
+                    .requestURL("https://sub1.example.com:10003/")
```

Comment:
I don't agree with this test.  We should be order independent for such headers.  If we decide that `X-Forwarded-Port` takes precedent over any port specified in `X-Forwarded-Host` then it should do so regardless of the order of those headers!

Reply:
The code in ForwardedRequestCustomizer is most definitely order dependent, esp around host/port/authority.  (IIRC, these are not the only tests that show this order dependency, i'll point out which other ones we have tomm.).
This order dependency was introduced when the code changed to iterating the fields from querying the fields.

If we want to change this code to not be order dependent, then we should go over all of the tests and expectations first, discuss them, then fix the code to conform to the expectations.

Reply:
I opened Issue #5247, proposing a better way.